cads_adaptors/adaptors/multi.py

```diff
@@ -41,43 +41,62 @@ def split_request(
 
         return this_request
 
-    def _pre_retrieve(self, request, default_download_format="zip"):
-        self.input_request = deepcopy(request)
-        self.receipt = request.pop("receipt", False)
-        self.download_format = request.pop("download_format", default_download_format)
-
-    def retrieve(self, request: Request):
+    def split_adaptors(
+        self, request: Request
+    ) -> dict[str, tuple[AbstractCdsAdaptor, Request]]:
         from cads_adaptors.tools import adaptor_tools
 
-        self._pre_retrieve(request, default_download_format="zip")
-
-        these_requests = {}
-        exception_logs: dict[str, str] = {}
-        self.context.logger.debug(f"MultiAdaptor, full_request: {request}")
+        sub_adaptors = {}
         for adaptor_tag, adaptor_desc in self.config["adaptors"].items():
-            this_adaptor = adaptor_tools.get_adaptor(adaptor_desc, self.form)
+            this_adaptor = adaptor_tools.get_adaptor(
+                {**adaptor_desc},
+                self.form,
+            )
+            # Set the sub-adaptor context to the MultiAdaptor context
+            this_adaptor.context = self.context
```

Comment:
This should work, right? If yes, I think it's better, especially if the context is or will be used at initialisation.
```suggestion
            this_adaptor = adaptor_tools.get_adaptor(
                adaptor_desc | {"context": self.context},
                self.form,
            )
```